src/drivers/indexeddb.js

```diff
@@ -181,27 +182,164 @@
             }
         }
 
+        // Initialize a singleton container for all running localForages.
+        if (!dbContexts) {
+            dbContexts = {};
+        }
+
+        // Get the current context of the database;
+        var dbContext = dbContexts[dbInfo.name];
+
+        // ...or create a new context.
+        if (!dbContext) {
+            dbContext = {
+                // Running localForages sharing a database.
+                forages: [],
+                // Shared database.
+                db: null
+            };
+            // Register the new context in the global container.
+            dbContexts[dbInfo.name] = dbContext;
+        }
+
+        // Register itself as a running localForage in the current context.
+        dbContext.forages.push(this);
+
+        // Create an array of readiness of the related localForages.
+        var ready = [];
+        for (var j = 0; j < dbContext.forages.length; j++) {
+            var forage = dbContext.forages[j];
+            if (forage !== this) { // Don't wait for itself...
+                ready.push(forage.ready());
+            }
+        }
+
+        // Take a snapshot of the related localForages.
+        var forages = dbContext.forages.slice(0);
+
+        // Initialize the connection process only when
+        // all the related localForages are ready.
+        return Promise.all(ready).then(function() {
+            dbInfo.db = dbContext.db;
+            // Get the connection or open a new one without upgrade.
+            return _getOriginalConnection(dbInfo);
+        }).then(function(db) {
+            dbInfo.db = db;
+            if (_isUpgradeNeeded(dbInfo, self._defaultConfig.version)) {
+                // Reopen the database for upgrading.
+                return _getUpgradedConnection(dbInfo);
+            }
+            return db;
+        }).then(function(db) {
+            dbInfo.db = dbContext.db = db;
+            self._dbInfo = dbInfo;
+            // Share the final connection amongst related localForages.
+            for (var k in forages) {
+                var forage = forages[k];
+                if (forage !== self) { // Self is already up-to-date.
+                    forage._dbInfo.db = dbInfo.db;
+                    forage._dbInfo.version = dbInfo.db;
+                }
+            }
+        });
```

Comment:
I admit it's a bit confusing, but "restoring the chain" doesn't mean changing the status of the original promise. And thankfully so, otherwise one can't be able to start different (and concurrent) chains on the same promise without massive side-effects between them. Promises are only, and exclusively, resolved or rejected by the executor given at their initialization.

Here is a [jsbin](https://jsbin.com/wujaki/edit?js,console) which illustrates this. It's based on the MDN example above and completed with our two arrays of promises. As one can see, even though `catch()` restores all chains in which it's been involved, only the second array `pAry2` is actually resolved.